packages/standalone/src/routes/ColumnProfile.svelte

```diff
@@ -16,64 +16,92 @@
 
     async function getPlot() {
         if (type == 'DOUBLE' || type == 'BIGINT') {
+            const data = await coordinator().query(`
+                SELECT 
+                    CASE 
+                        WHEN MIN("${colName}") = MAX("${colName}") THEN 'same'
+                        ELSE 'distinct'
+                    END AS result
+                FROM dataprof
+            `, { cache: false });
+            //@ts-ignore
+            const res = Array.from(data)[0].result;
+
             const element = document.querySelector(`#${uniqueId}`);
             if (element) {
-                element.replaceChildren(vg.vconcat(vg.plot(
-                    vg.rectY(
-                        vg.from("dataprof", { filterBy: brush }),
-                        { x: vg.bin(colName), y: vg.count(), fill: "steelblue", inset: 0.5 }
-                    ),
-                    vg.intervalX({ as: brush }),
-                    vg.xDomain(vg.Fixed),
-                    vg.yTickFormat("s"),
-                    vg.width(500),
-                    vg.height(80)
-                )));
+                if(res === 'distinct'){
+                    element.replaceChildren(vg.vconcat(vg.plot(
+                        vg.rectY(
+                            vg.from("dataprof", { filterBy: brush }),
+                            { x: vg.bin(colName), y: vg.count(), fill: "steelblue", inset: 0.5 }
+                        ),
+                        vg.intervalX({ as: brush }),
+                        vg.xDomain(vg.Fixed),
+                        vg.yTickFormat("s"),
+                        vg.width(500),
+                        vg.height(80)
+                    )));
+                } else {
+                    element.replaceChildren(vg.vconcat(vg.plot(
+                        vg.rectY(
+                            vg.from("dataprof", { filterBy: brush }),
+                            { x: colName, y: vg.count(), fill: "steelblue", inset: 0.5 }
+                        ),
+                        vg.intervalX({ as: brush }),
+                        vg.xDomain(vg.Fixed),
+                        vg.yTickFormat("s"),
+                        vg.width(500),
+                        vg.height(80)
+                    )));
+                }
             }
         } else if (type == 'DATE') {
-            const countOfEachDate = await coordinator().query(`
-                SELECT ${colName}, COUNT(*) AS count
+            const data = await coordinator().query(`
```

Comment:
Can we use https://github.com/uwdata/mosaic/pull/435 instead?

Reply:
I kept getting the parser error with the query when I set the dates to be automatically binned. I think it runs into the error if there is a difference of over 40 years between the max and min dates, so I used the auto-binning if the difference is <= 40 years and otherwise, I manually set the bin to have an interval of years and a step of 1.

Reply:
Let's dig into this and see whether it's a bug in mosaic. It's always best to fix issues upstream.